ts/src/pro/poloniex.ts

```diff
@@ -0,0 +1,971 @@
+            //        },
+            //        "channel": "auth"
+            //    }
+            //
+            client.subscriptions[messageHash] = future;
+        }
+        return future;
+    }
+
+    async subscribe (name: string, isPrivate: boolean, symbols: string[] = undefined, params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @description Connects to a websocket channel
+         * @param {string} name name of the channel
+         * @param {boolean} isPrivate true for the authenticated url, false for the public url
+         * @param {[string]|undefined} symbols CCXT market symbols
+         * @param {object} params extra parameters specific to the poloniex api
+         * @returns {object} data from the websocket stream
+         */
+        const publicOrPrivate = isPrivate ? 'private' : 'public';
+        const url = this.urls['api']['ws'][publicOrPrivate];
+        const subscribe = {
+            'event': 'subscribe',
+            'channel': [
+                name,
+            ],
+        };
+        const marketIds = [ ];
+        let messageHash = name;
+        if (symbols !== undefined) {
+            if (symbols.length === 1) {
+                const symbol = symbols[0];
+                const marketId = this.marketId (symbol);
+                marketIds.push (marketId);
+                messageHash = messageHash + ':' + symbol;
+            }
+            for (let i = 0; i < symbols.length; i++) {
+                const symbol = symbols[i];
+                marketIds.push (this.marketId (symbol));
+            }
+        } else {
+            marketIds.push ('all');
+        }
+        if (name !== 'balances') {
+            subscribe['symbols'] = marketIds;
+        }
+        const request = this.extend (subscribe, params);
+        return await this.watch (url, messageHash, request, name);
+    }
+
+    async watchOHLCV (symbol: string, timeframe = '1m', since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name poloniex#watchOHLCV
+         * @description watches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @see https://docs.poloniex.com/#public-channels-market-data-candlesticks
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe the length of time each candle represents
+         * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
+         * @param {int|undefined} limit the maximum amount of candles to fetch
+         * @param {object} params extra parameters specific to the poloniex api endpoint
+         * @returns [[int]] A list of candles ordered as timestamp, open, high, low, close, volume
+         */
+        await this.loadMarkets ();
+        const channel = this.safeString (this.timeframes, timeframe, timeframe);
```

Comment:
First, `this.timeframes` have the rest timeframes, afaik right now, we can't override it automatically. Second, even if we could we shouldn't because a pro instance might be used to perform rest calls.

Reply:
Will move them inside options for now